$OpenBSD: patch-src_main_c,v 1.1 2011/01/05 14:06:51 dcoppa Exp $

Fix high cpu usage if stdin is /dev/null (i.e. when started with "nohup")

--- src/main.c.orig	Sat Nov  6 13:38:14 2010
+++ src/main.c	Wed Jan  5 14:46:15 2011
@@ -317,6 +317,10 @@ int main (int argc, char **argv) {
 				curFd = ctlFd;
 			}
 			buf = fgetc (curFd);
+			if (buf == EOF) {
+			  /* select() is going wild if fdset contains EOFed fd's */
+			  FD_CLR (fileno (curFd), &readSet);
+			}
 
 			size_t i;
 			for (i = 0; i < BAR_KS_COUNT; i++) {
